### ISSUE TITLE
Dropdown menus badly displayed when the button is on the right of the screen

### ISSUE THREAD

**Issue by gdelhumeau** [twbs/bootstrap#16968]
Hello.

The dropdown menu is not displayed right under the button when there is not enough horizontal place on the screen.

See an example:
```
<div class="dropdown text-right">
  <a href="#" data-toggle="dropdown" class="dropdown-toggle" role="button">Click on me</a>
  <ul class="dropdown-menu">
    <li><a href="#">Action</a></li>
    <li><a href="#">Action</a></li>
  </ul>
</div>
```
You can see the results on: http://jsfiddle.net/uywss44p/1/

![Screenshot of the problem](http://xwiki.kephpage.net/bootstrap/issue-dropdown/bad.png)

In my opinion, when there is not enough place on the right side of the screen, the dropdown menu should still be displayed under the button, with the top-right corner aligned with the bottom-right corner of the button, exactly as it is on a right centered navbar (see http://jsfiddle.net/uywss44p/3/).

![Screenshot of a good behaviour](http://xwiki.kephpage.net/bootstrap/issue-dropdown/good2.png)

Thanks,

Guillaume

**Comment by twbs-lmvtfy** [comment]
Hi @gdelhumeau!

You appear to have posted a live example (http://fiddle.jshell.net/uywss44p/2/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 39, column 3: [W012](https://github.com/twbs/bootlint/wiki/W012): `.container` or `.container-fluid` should be the first child inside of a `.navbar`

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by gdelhumeau** [comment]
I've fixed the example as proposed by the bot, but it does not concern the issue which is still valid.

**Comment by gdelhumeau** [comment]
Actually the behaviour is not the same on my use-case:
![Screenshot of the problem](http://xwiki.kephpage.net/bootstrap/issue-dropdown/bad2.png)

On this situation: replacing adding `right: 0` is enough to fix the problem, but it must be done in the javascript code only where the place is not enough.

![Screenshot of the fixed situation](http://xwiki.kephpage.net/bootstrap/issue-dropdown/good3.png)

**Comment by gdelhumeau** [comment]
For the record, I've used this fix on my project:
```
require(['jquery', 'bootstrap'], function($) {
  // TODO: remove this when https://github.com/twbs/bootstrap/issues/16968 is fixed
  $('.dropdown').on('shown.bs.dropdown', function () {
    var menu = $(this).find('.dropdown-menu');
    var menuLeft = menu.offset().left;
    var menuWidth = menu.outerWidth();
    var documentWidth = $(body).outerWidth();
    if (menuLeft + menuWidth > documentWidth) {
      menu.offset({'left': documentWidth - menuWidth});
    }
  })
});
```
It works on my use-case that I did not manage to reproduce on fiddle.

**Comment by cvrebert** [comment]
Approximate duplicate of #10756.
